### PR TITLE
Add dotnet-maui plugin with dotnet-maui-doctor skill

### PR DESCRIPTION
**Pull request by jfversluis**
Closes #14

## Summary

Introduces the `dotnet-maui` plugin and moves the `dotnet-maui-doctor` skill (originally from #14 by @Redth) into the correct `plugins/` directory structure.

### What's included

- **`plugins/dotnet-maui/plugin.json`** — New plugin manifest
- **`plugins/dotnet-maui/skills/dotnet-maui-doctor/`** — Skill + 11 reference docs covering:
  - Environment detection (macOS, Windows, Linux)
  - .NET SDK, workload, JDK, Android SDK, Xcode, and Windows SDK validation
  - Dynamic version discovery via NuGet WorkloadDependencies.json (no hardcoded versions)
  - Platform-specific installation commands (Bash + PowerShell)
  - Troubleshooting guides
- **`tests/dotnet-maui/dotnet-maui-doctor/eval.yaml`** — 8 eval scenarios covering all platforms, guardrails, and edge cases
- **Updated**: `marketplace.json`, `CODEOWNERS`, `README.md`

### Changes from #14

- Moved skill from `.agents/skills/` to `plugins/dotnet-maui/skills/` per @timheuer's feedback
- Added `When to Use` / `When Not to Use` / `Inputs` / `Common Pitfalls` sections to SKILL.md
- Added PowerShell equivalents for all Bash commands in workload-dependencies-discovery.md
- Added `jq`/`unzip` as explicit prerequisites
- Replaced hardcoded JDK 21 workaround with dynamic WorkloadDependencies.json guidance
- Created 8 eval scenarios for the skill validator

### Eval scenarios

| # | Scenario | Tests |
|---|----------|-------|
| 1 | macOS setup with Xcode | Xcode from Developer Downloads, `maui` workload, MS OpenJDK |
| 2 | Linux Android-only | `maui-android` not `maui`, no iOS workloads |
| 3 | Workload update/repair guardrail | Refuses update/repair, uses `install --version` |
| 4 | Non-Microsoft JDK diagnosis | Identifies vendor (not version) as problem |
| 5 | Windows full setup | `sdkmanager.bat`, PowerShell commands, Windows SDK |
| 6 | JAVA_HOME misconception | Auto-detection, JAVA_HOME not required |
| 7 | Android SDK package discovery | CI-oriented manifest-driven package requirements |
| 8 | Stale workloads after SDK update | Version-aligned reinstall, not update/repair |

### Note

The `eval.yaml` has not yet been run through the skill validator (`eng/skill-validator`). This requires .NET 10 SDK and authenticated GitHub access. Happy to run it once the infrastructure is available or if a maintainer can assist.

Continues work from #14 by @Redth — his original commits are preserved.

cc @Redth @timheuer @jeffschwMSFT

### CODE REVIEW

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+# Linux: ~/Android/Sdk
+# Windows: %LOCALAPPDATA%\Android\Sdk
+
+# Check if sdkmanager is available
+# macOS/Linux
+$ANDROID_SDK_ROOT/cmdline-tools/latest/bin/sdkmanager --version
+
+# Windows
+%ANDROID_SDK_ROOT%\cmdline-tools\latest\bin\sdkmanager.bat --version
+```
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md:108]: Fixed — replaced `%ANDROID_SDK_ROOT%` cmd.exe syntax with PowerShell `$env:ANDROID_SDK_ROOT`.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-macos.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+```bash
+# Check if any simulators already exist
+xcrun simctl list devices available
+
+# If none exist, find the latest iPhone device type and runtime
+xcrun simctl list devicetypes | grep iPhone
+xcrun simctl list runtimes | grep iOS
+
+# Create one using the newest iPhone type and latest iOS runtime
+xcrun simctl create "iPhone 16 Pro" "com.apple.CoreSimulator.SimDeviceType.iPhone-16-Pro" "com.apple.CoreSimulator.SimRuntime.iOS-18-0"
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-macos.md]: Fixed — replaced hardcoded device type and runtime with placeholders.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/workload-dependencies-discovery.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+**PowerShell:**
+```powershell
+Invoke-WebRequest "https://api.nuget.org/v3-flatcontainer/{packageid}/{version}/{packageid}.{version}.nupkg" -OutFile manifest.nupkg
+$zip = [System.IO.Compression.ZipFile]::OpenRead("manifest.nupkg")
+$entry = $zip.Entries | Where-Object { $_.FullName -eq "data/WorkloadDependencies.json" }
+$reader = [System.IO.StreamReader]::new($entry.Open())
+$reader.ReadToEnd() | ConvertFrom-Json
+$reader.Dispose(); $zip.Dispose()
+```
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/workload-dependencies-discovery.md:117]: Fixed — added `Add-Type -AssemblyName System.IO.Compression.FileSystem` before the ZipFile call.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+### "Workload version mismatch"
+
+**Cause**: Workloads from different SDK versions or incomplete installation.
+
+**Solution**: Reinstall workloads with explicit version:
+```bash
+# First, find the correct workload set version for your SDK
+# Query NuGet APIs (see workload-dependencies-discovery.md)
+
+# Then reinstall with explicit version
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md]: Fixed — added clarifying comment that ios/maccatalyst should be omitted on Linux.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+dotnet workload install android --version $WORKLOAD_VERSION
+```
+
+### "Could not find android.jar"
+
+**Cause**: Android platform not installed.
+
+**Solution**:
+```bash
+$ANDROID_SDK_ROOT/cmdline-tools/latest/bin/sdkmanager "platforms;android-35"
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md]: Fixed — replaced hardcoded `android-35` with `$API_LEVEL` placeholder referencing WorkloadDependencies.json.

File: plugins/dotnet-maui/plugin.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "dotnet-maui",
+  "version": "0.1.0",
+  "description": "Skills for .NET MAUI development: environment setup, diagnostics, and troubleshooting.",
+  "skills": "./skills/"
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply plugins/dotnet-maui/plugin.json:5]: Intentionally omitted — this plugin has no agents, and adding an empty `agents/` directory with a dangling path in the manifest would be misleading. The other plugins that include `agents` actually have agent definitions in that folder. If/when we add agents to this plugin, we'll add both the directory and the manifest entry.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-macos.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
+# macOS Installation Commands
+
+## Xcode
+
+### Install Xcode
+
+**Do not install Xcode from the App Store** — it can auto-update to a version newer than what .NET MAUI supports.
+
+Download a specific version from [Apple Developer Downloads](https://developer.apple.com/download/all/), matching the `xcode.version` range from WorkloadDependencies.json.
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-macos.md:9]: Does something need to mention you will need to manually login with your Apple account? (it has 2FA)

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-windows.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+## Windows SDK
+
+The Windows SDK is required for WinUI 3 / Windows targets.
+
+### Detect Windows SDK
+
+```powershell
+Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows Kits\Installed Roots" -ErrorAction SilentlyContinue
+```
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-windows.md]: Is this the right command, it just reports random things for me:
```
> Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows Kits\Installed Roots" -ErrorAction SilentlyContinue


{599B5A52-33FD-0EA6-D7F2-E62F2058A21C} : Application Verifier x64 External Package (DesktopEditions) x64
KitsRoot10                             : C:\Program Files (x86)\Windows Kits\10\
{9624EAC7-C752-8DC3-33D7-6911F0889062} : Application Verifier x64 External Package (OnecoreUAP) x64
AppVerifier64BitAutomationRoot         : C:\Program Files\Application Verifier\
{ACB8119F-54A6-C93C-CA2F-3A1476EE8869} : Application Verifier x64 External Package (DesktopEditions) x64
{C53ECC06-72B1-FEBF-D241-038A4BC83E57} : Application Verifier x64 External Package (OnecoreUAP) x64
{A6FC8F56-AEE7-CC1C-28D6-A17567A48D75} : Application Verifier x64 External Package (DesktopEditions) x64
{27C8EF42-3846-AFCF-2F6D-8C145F75C8EF} : Application Verifier x64 External Package (OnecoreUAP) x64
{A891E3DB-0F4D-B1C1-EC88-CF1C8ABAF8CB} : Application Verifier x64 External Package (DesktopEditions) x64
{D9009790-F40A-C423-CCB8-65ECEC9A9FA9} : Application Verifier x64 External Package (OnecoreUAP) x64
{AAD4ECCF-69C7-CA36-E72C-BB00821FB9FA} : Application Verifier x64 External Package (DesktopEditions) x64
{F0B26EED-55C0-CCDD-F474-34FFFD3F3AB2} : Application Verifier x64 External Package (OnecoreUAP) x64
{43D478B3-5651-284E-0CA3-C3AC2C1CDFC9} : Application Verifier x64 External Package (DesktopEditions) x64
{6A220D0F-7788-4295-2445-FF790A88060D} : Application Verifier x64 External Package (OnecoreUAP) x64
{C0B0A1D9-9045-54FE-86F6-1B0F836E665E} : Application Verifier x64 External Package (DesktopEditions) x64
{674C8368-9677-55A8-FAD9-A86A3073E60D} : Application Verifier x64 External Package (OnecoreUAP) x64
{D75B3C63-F3BB-5EC1-45A7-15C00BBB9D3F} : Application Verifier x64 External Package (DesktopEditions) x64
{601DDF69-FCF4-F568-946A-C17046C946E4} : Application Verifier x64 External Package (OnecoreUAP) x64
{20C01991-CCD1-2C06-7A9A-B10A9B4AF807} : Windows App Certification Kit Native Components x64
{C7F2EF85-ACD3-0480-6BC2-0C3B45A8F1C1} : Universal CRT Tools x64 x64
{FB7C4DC7-7641-04B7-6639-589757B86783} : Windows SDK DirectX x64 Remote x64
PSPath                                 : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows Kits\Installed Roots
PSParentPath                           : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows Kits
PSChildName                            : Installed Roots
PSDrive                                : HKLM
PSProvider                             : Microsoft.PowerShell.Core\Registry
```

Some are Windows SDK things, just worth checking if this the official way.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+  jq -r '.resources[] | select(.["@type"]=="SearchQueryService") | .["@id"]' | head -1)
+
+# Query for latest workload set
+# SDK band = first 2 segments of SDK version (e.g., 10.0 from 10.0.102)
+curl -s "$NUGET_SEARCH_URL?q=Microsoft.NET.Workloads.$SDK_BAND&prerelease=false" | \
+  jq '.data[] | select(.id | test("^Microsoft.NET.Workloads.$SDK_BAND.[0-9]+$")) | {id, version}'
+
+# Convert NuGet version to CLI version:
+# NuGet A.B.C → CLI A.0.B (e.g., NuGet 10.102.0 → CLI 10.0.102)
+```
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md:33]: This looks like a manual bash version of the `dotnet workload search version 10.0.103` command, can AI use that instead?

If that doesn't work, can we file an issue on dotnet/sdk?

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+| Set to non-Microsoft JDK | ❌ | Unset it or point to Microsoft JDK |
+
+To unset:
+```bash
+# macOS/Linux
+unset JAVA_HOME
+
+# Windows PowerShell
+Remove-Item Env:JAVA_HOME
+```
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md]: `JAVA_HOME` isn't required, but I don't think we have to tell AI to _unset_ it?

Maybe it could just report this as an "anomoly" if it finds an unexpected JDK in `$JAVA_HOME`?

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting-windows.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+## Windows Diagnostic Commands
+
+```powershell
+# Windows SDK detection
+Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows Kits\Installed Roots" -ErrorAction SilentlyContinue
+
+# JDK detection (Windows-specific)
+Get-ChildItem "C:\Program Files\Microsoft" -Filter "jdk-*" -ErrorAction SilentlyContinue
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting-windows.md]: can this use `$env:ProgramFiles`

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+
+```
+C:\Program Files\Microsoft\jdk-{VERSION}\
+```
+
+Registry: `HKLM\SOFTWARE\Microsoft\JDK\{VERSION}`
+
+Detection:
+```powershell
+Get-ChildItem "C:\Program Files\Microsoft" -Filter "jdk-*" -ErrorAction SilentlyContinue
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md]: `$env:ProgramFiles`

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md [hunk 13]
```diff
@@ -0,0 +1,317 @@
...
+
+# Option 2: Point to Microsoft JDK (macOS)
+export JAVA_HOME=/Library/Java/JavaVirtualMachines/microsoft-{VERSION}.jdk/Contents/Home
+
+# Option 2: Point to Microsoft JDK (Linux)
+export JAVA_HOME=/usr/lib/jvm/msopenjdk-{VERSION}
+
+# Option 2: Point to Microsoft JDK (Windows PowerShell)
+$env:JAVA_HOME = "C:\Program Files\Microsoft\jdk-{VERSION}"
+# Or remove it: Remove-Item Env:JAVA_HOME
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md]: I'm confused why we remove and re-set `$JAVA_HOME`, maybe it's better to just report something is wrong. An "anamoly" if we find a different JDK in `$JAVA_HOME`.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md [hunk 14]
```diff
@@ -0,0 +1,317 @@
...
+**Cause**: Oracle, Azul, Corretto, or other non-Microsoft JDK is installed and selected.
+
+**How to identify**: Run `java -version` - if output does NOT contain "Microsoft", wrong JDK is selected.
+
+**Solution**:
+1. Install the recommended Microsoft OpenJDK version (see commands above)
+2. Set `JAVA_HOME` to Microsoft JDK path:
+   - macOS: `/Library/Java/JavaVirtualMachines/microsoft-{VERSION}.jdk/Contents/Home`
+   - Windows: `C:\Program Files\Microsoft\jdk-{VERSION}\`
+   - Linux: `/usr/lib/jvm/msopenjdk-{VERSION}`
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md]: I don't think we really ever need to set `$JAVA_HOME`. Maybe an example of what this fixes would help me understand?

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+---
+
+## Android SDK
+
+### Detecting Existing Android SDK
+
+```bash
+# Check common environment variables
+echo $ANDROID_HOME
+echo $ANDROID_SDK_ROOT
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md:91]: We never set these, and neither does Android Studio.

Is the purpose to just find out if a user has set these themselves?

Comment (jonathanpeppers) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md:91]: `ANDROID_HOME` is also deprecated: https://developer.android.com/tools/variables#envar

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-windows.md (hunk 8 above)
Comment (Redth) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-windows.md]: I was debating removing this windows SDK part already, maybe we can just leave it out for now.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md (hunk 15 above)
Comment (Redth) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md:91]: Should we expand a bit to articulate known paths this might exist instead?  I agree, probably checking for these isn't super useful outside of may CI/CD environments.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md (hunk 10 above)
Comment (Redth) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md]: Yeah perhaps just a bit of guidance so that if it detects a non-MS OpenJDK being set to that, only then do we bother trying to unset it.  Though, if we aren't even checking it, does it matter if it's set?

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md (hunk 13 above)
Comment (Redth) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md]: @jonathanpeppers do we ever look at JAVA_HOME for our own tooling to find a JDK?  If there's no registry key set, no Monodroid config set, etc... Do we just search known paths and not check JAVA_HOME? I think if that is truly the case, then yes, we could remove anything suggesting to set / unset JAVA_HOME and just forget about that part.  Actually we probably would need to do the opposite and note to _never_ care about JAVA_HOME and what it's set to.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-macos.md (hunk 7 above)
Comment (Redth) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-macos.md:9]: Probably worth adding a note so it can tell the developer what to expect.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md (hunk 9 above)
Comment (Redth) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md:33]: I believe this would be running `dotnet workload search version 10.0.100` since it will have determined the major sdk version's "band" would be 10.0.100 since the sdk could be 10.0.102, 10.0.103, etc.

If you search with the sdk version exactly, you'll get exactly that manifest version back, not anything newer, which I am not sure is exactly the result we want? Though it could be ok I suppose.

Though looking at the command it does seem that `dotnet workload search version --format json` will return the newest N (you can use --take to scope it to 1 too) workload set versions.  This is maybe more what we want.

```
> dotnet workload search version --format json --take 1
[{"workloadVersion":"10.0.103"}]
```

That would eliminate the need to look up a the NUGET_SEARCH_URL and then the subsequent request for the versions of the workload set package.

So yeah, I think this would be a good simplification.  It could be updated in this file as well as the workload-dependencies-discovery.md file.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md (hunk 13 above)
Comment (jonathanpeppers) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md]: It will look for `$JAVA_HOME` before looking for hardcoded/known paths, but nothing sets this env var -- a developer would have to set it themselves.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-macos.md (hunk 7 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-macos.md:9]: Added a note about Apple ID + 2FA requirement. The agent will now tell the developer they need to manually log in and download Xcode, then continue with the remaining steps.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-windows.md (hunk 8 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-windows.md]: Removed the Windows SDK detection commands entirely — per Redth's suggestion, the Windows SDK is installed automatically as part of the MAUI workload/Visual Studio and doesn't need manual detection.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md (hunk 9 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md:33]: Great suggestion! Replaced the NuGet API approach with `dotnet workload search version --format json --take 1` as Redth also confirmed this is the right approach. Updated both installation-commands.md and workload-dependencies-discovery.md.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md (hunk 10 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md]: Agreed. Reframed to report as anomaly rather than prescribing unset/set commands. The skill will flag a non-Microsoft JDK in JAVA_HOME and let the user decide what to do.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting-windows.md (hunk 11 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting-windows.md]: Done — using `$env:ProgramFiles` now.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md (hunk 12 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md]: Fixed — using `$env:ProgramFiles` instead of hardcoded path.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md (hunk 13 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md]: Reframed: the skill now reports a non-Microsoft JDK in JAVA_HOME as an anomaly and lets the user decide, rather than prescribing unset/set commands. Thanks for the clarification that the tooling checks JAVA_HOME first, then falls back to known paths.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md (hunk 14 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md]: You're right — removed the guidance to set JAVA_HOME. The skill now just reports if it finds a non-Microsoft JDK there and lets the user decide.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md (hunk 15 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md:91]: Expanded with known filesystem paths per platform (~/Library/Android/sdk, ~/Android/Sdk, etc.) so the agent can check those directly instead of relying only on env vars.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-windows.md (hunk 8 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-windows.md]: Done — removed the detection commands. The file now just notes the Windows SDK is installed via MAUI workload/VS Installer with a link to standalone install if needed.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md (hunk 15 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md:91]: Added known filesystem paths (~/Library/Android/sdk on macOS, ~/Android/Sdk on Linux, $env:LOCALAPPDATA\Android\Sdk on Windows) so the agent can check those directly. Env vars are still listed as a first check.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md (hunk 10 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md]: Per jonathanpeppers' confirmation that the tooling checks JAVA_HOME first before known paths: reframed the guidance to report it as an anomaly if a non-Microsoft JDK is found there, rather than prescribing unset/set. The skill won't proactively mess with JAVA_HOME.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md (hunk 13 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/troubleshooting.md]: Based on jonathanpeppers' answer: the tooling does check JAVA_HOME first, then falls back to known paths. Nothing auto-sets it though. Updated to treat non-Microsoft JAVA_HOME as an anomaly — the skill reports it but doesn't auto-modify the environment.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-macos.md (hunk 7 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands-macos.md:9]: Added a note explaining that Apple Developer Downloads requires signing in with an Apple ID (with two-factor authentication), and the agent should tell the developer to download Xcode manually.

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md (hunk 9 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/installation-commands.md:33]: Great feedback! Updated both installation-commands.md and workload-dependencies-discovery.md to use `dotnet workload search version --format json --take 1`. This eliminates the NuGet search API step entirely — much simpler. The NuGet download URLs are still used for Steps 3-4 (manifest extraction for WorkloadDependencies.json).

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+
+## Identifying Microsoft OpenJDK
+
+Run `java -version`. Microsoft OpenJDK output contains `Microsoft` in the runtime line:
+
+```
+openjdk version "17.0.14" 2025-01-21 LTS
+OpenJDK Runtime Environment Microsoft-XXXXXXX (build 17.0.14+7-LTS)
+OpenJDK 64-Bit Server VM Microsoft-XXXXXXX (build 17.0.14+7-LTS, mixed mode, sharing)
+```
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md:22]: Can we put a JDK 21 example?

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/workload-dependencies-discovery.md [hunk 17]
```diff
@@ -0,0 +1,249 @@
...
+### Step 5: Parse WorkloadDependencies.json
+
+**Android workload** (`microsoft.net.sdk.android`):
+```json
+{
+  "microsoft.net.sdk.android": {
+    "jdk": {
+      "version": "[17.0,22.0)",
+      "recommendedVersion": "17.0.14"
+    },
```
Comment (jonathanpeppers) [inline plugins/dotnet-maui/skills/dotnet-maui-doctor/references/workload-dependencies-discovery.md:128]: I'm going to check if we are missing an update to get this in 10, but can we put:

```suggestion
    "jdk": {
      "version": "[17.0,22.0)",
      "recommendedVersion": "21.0.8"
    },
```

https://dev.azure.com/devdiv/DevDiv/_git/android-platform-support?path=/src/Xamarin.Installer.Build.Tasks/Xamarin.Installer.Common.props&version=GBmain&line=10&lineEnd=10&lineStartColumn=5&lineEndColumn=82&lineStyle=plain&_a=contents

File: plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md (hunk 16 above)
Comment (jfversluis) [reply plugins/dotnet-maui/skills/dotnet-maui-doctor/references/microsoft-openjdk.md:22]: Updated the example to JDK 21. Good catch — .NET 10 MAUI uses JDK 21 now.
